pkg/controller/seed-controller-manager/cni-application-installation-controller/controller.go

```diff
@@ -365,6 +365,13 @@ func ApplicationInstallationReconciler(cluster *kubermaticv1.Cluster, overwriteR
 				ipam := values["ipam"].(map[string]any)
 				operator := ipam["operator"].(map[string]any)
 				delete(operator, "clusterPoolIPv4PodCIDR")
+
+				// If not specified, set envoy.enabled to false
+				// https://github.com/cilium/cilium/commit/471f19a16593e1e9342c31bf3e26e5383737cb0a
+				envoy := values["envoy"].(map[string]any)
+				if _, ok := envoy["enabled"]; !ok {
+					envoy["enabled"] = false
+				}
```

Comment:
Instead of this, let's address the issue directly by modifying/fixing the problematic `ApplicationDefinition`. That will improve clarity and ensure consistency across all KKP 2.27 installations, including new and upgraded ones.